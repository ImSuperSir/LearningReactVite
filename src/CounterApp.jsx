import { useState }  from 'react'; // "use" por definiciÃ³n es usado para identificar los hooks
import PropTypes from 'prop-types';


export const CounterApp  = ({value}) => {


  const [counter, setCounter] = useState(value);  //los hooks deben ser constantes, las constantes son mas ligeras

  // function handleAdd(event) {
  //   ...
  // } ;
  const handleAdd = (event) => {
    /* de aca para el manejo del evento, podemos desestructura y
    o realizar cualquier cosa que deseemos */
    // console.log(event)
    //setCounter(counter + 1);  
                              /*practicamente le estamos diciento que hay un cambio en el estado, y el componente se redibuja
                              No podemos realizar counter++, ya que counter es tipo const, y eso es correcto
                              lo que hacemos con el setcounter es que redespliega el componente. */

      setCounter( (c) => c + 1) ; 
  }

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{ counter }</h2>
      {/* <button onClick={ (event) => handleAdd(event)}> */}
      <button onClick={ handleAdd }>  {
                                        /* "handleAdd" esta linea esta simplificada, porque 
                                        solo se maneja un argumento, 
                                        que lo toma por default (checar la linea anterior),
                                        osea es lo mismmo que: (event) => handleAdd(event)  
                                        El evento onclick recibe el event, y este lo envia a handleAdd
                                        */}
        +1
      </button>
    </>
  );
}

CounterApp.propTypes = {
  value : PropTypes.number.isRequired,
}


