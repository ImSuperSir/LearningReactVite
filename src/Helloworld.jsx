// export function App()
// {
//   return (<h1>Hola Laurito Ramirez!!!</h1>);
// }

import React from 'react'

// export const Helloworld = () => {
//   return (
//     <div>
//       Hola Lauroa Ramirez Div      
//     </div>
//   )
// }

export const Helloworld = () => (
<div>Hola Mundo... super simplificado..</div>

);

//export default App;
