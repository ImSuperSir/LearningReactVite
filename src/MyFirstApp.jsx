

//Esta es mi primera aplicacion

export const MyFirstApp = () => (<h1>Esta es mi primera aplicaciÃ³n</h1>);