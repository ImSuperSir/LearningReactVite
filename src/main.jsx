import React, { StrictMode } from 'react';
import ReactDOM from 'react-dom/client';
// import { Helloworld } from './Helloworld.jsx'
import {MyFirstApp} from './MyFirstApp.jsx'

import './styles.css'





ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <MyFirstApp  title="Software Engineer" subtitle="Lauro Ramirez H."/> */}
    <MyFirstApp />
  </React.StrictMode>
);