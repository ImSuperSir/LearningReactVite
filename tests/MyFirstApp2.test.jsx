
// screen es lo que se renderea
import { getAllByText, render, screen } from "@testing-library/react"
import { MyFirstApp } from "../src/MyFirstApp"

describe('Test on <MyFirstApp />', () =>
{

  const lTitle = 'Hola Im Super Sir';
  const lSubTitle = "Software engineer";

  test('should match with the snapshot', () =>
  {

    const { container } = render(<MyFirstApp title={lTitle} />);
    expect(container).toMatchSnapshot();
  });

  test('should display the message "Hola Im Super Sir"', () =>
  {
    render(<MyFirstApp title={lTitle} />);
    screen.debug(); //todo el html generado de nuestro componente renderizado

    expect(screen.getByText(lTitle)).toBeTruthy();

  });

  test('should display the title into a h1 tag', () =>
  {

    render(<MyFirstApp title={lTitle} />);
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(lTitle);

  });

  test('should display the subtitle', () =>
  {

    render(
      <MyFirstApp
        title={lTitle}
        subtitle={lSubTitle}
      />);

    expect(screen.getAllByText(lSubTitle).length).toBe(1);


  })



});
