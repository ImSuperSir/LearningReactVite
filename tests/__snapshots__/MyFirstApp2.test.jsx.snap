// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <MyFirstApp /> should match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola Im Super Sir
  </h1>
  <h2>
    Ingeniero de Software
  </h2>
  <p>
    Frida Sophia Ram√≠rez
  </p>
</div>
`;
